src/__rt/pinned_future.rs

```diff
@@ -27,11 +27,11 @@ impl<T> StableFuture for GenStableFuture<T>
     type Item = <T::Return as IsResult>::Ok;
     type Error = <T::Return as IsResult>::Err;
 
-    fn poll(mut self: Pin<Self>, ctx: &mut task::Context) -> Poll<Self::Item, Self::Error> {
+    fn poll(mut self: PinMut<Self>, ctx: &mut task::Context) -> Poll<Self::Item, Self::Error> {
```

Comment:
It should be  possible to factor out the `cell.set`s to be safe, too-- not just the gets.